pkg/daemons/agent/agent_windows.go

```diff
@@ -122,5 +133,77 @@ func kubeletArgs(cfg *config.Agent) map[string]string {
 	if cfg.ProtectKernelDefaults {
 		argsMap["protect-kernel-defaults"] = "true"
 	}
+
 	return argsMap
 }
+
+func waitForSourceVip(networkName string) string {
+	for range time.Tick(time.Second * 5) {
+		network, err := hcsshim.GetHNSNetworkByName(networkName)
+		if err != nil {
+			logrus.WithError(err).Warning("can't find HNS network, retrying", networkName)
+			continue
+		}
+		if network.ManagementIP == "" {
+			continue
+		}
+
+		script := `function GetSourceVip {
+			param(
+					$NetworkName
+				)
+			$hnsNetwork = Get-HnsNetwork | ? Name -EQ $NetworkName.ToLower()
```

Comment:
Can we not do this via `hcsshim.GetHNSNetworkByName` instead of inlining a bunch of powershell?

Reply:
Sure thing. We can fold it into pure Go.